lib/connect.ts

```diff
@@ -1341,38 +1341,38 @@ regular).
         return data;
     }
 
-    function _get(route, params, responseType, responseTransformer, isJSON = false) {
+    function _get(route, params?, responseType?, responseTransformer?, isJSON = false) {
         return request(route, "GET", params || {}, responseType, responseTransformer, isJSON);
     }
 
-    function _post(route, params, responseType, responseTransformer, isJSON = false, queryParams=null) {
+    function _post(route, params, responseType?, responseTransformer?, isJSON = false, queryParams: any =null) {
         return request(route, "POST", params || {}, responseType, responseTransformer, isJSON, queryParams);
     }
 
-    function _put(route, params, responseType, responseTransformer, isJSON = false, queryParams=null) {
+    function _put(route, params, responseType?, responseTransformer?, isJSON = false, queryParams=null) {
         return request(route, "PUT", params || {}, responseType, responseTransformer, isJSON, queryParams);
     }
 
-    function _delete(route, params, responseType, responseTransformer, isJSON = false) {
```

Comment:
One piece of Advice, we can make a custom hook like `useApi` to handle all these kinds of API methods.